core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1449,6 +1449,20 @@ class Partition(val topicPartition: TopicPartition,
     if (fetchParams.isFromFollower) {
       // Check that the request is from a valid replica before doing the read
       val (replica, logReadInfo) = inReadLock(leaderIsrUpdateLock) {
+        // For mirrored partitions, validate the follower's mirrorLeaderEpoch against the log's highest epoch
+        if (mirrorName.nonEmpty && fetchPartitionData.mirrorLeaderEpoch.isPresent) {
+          val followerMirrorEpoch = fetchPartitionData.mirrorLeaderEpoch.get()
+          val logHighestEpoch = localLogOrException.latestEpoch.orElse(0)
+
+          if (followerMirrorEpoch < logHighestEpoch) {
+            info(s"==== Fencing follower: partition=$topicPartition, followerMirrorEpoch=$followerMirrorEpoch, logHighestEpoch=$logHighestEpoch")
+            // Follower's mirrorLeaderEpoch is outdated
+            throw new FencedLeaderEpochException(
+              s"Follower's mirrorLeaderEpoch $followerMirrorEpoch is lower than leader's highest epoch $logHighestEpoch for mirrored partition $topicPartition"
```

Comment:
I think this change is not needed.

Reply:
I think it is needed for liveness, not for correctness.

This is what happens without leader-side fencing:

1. Follower skips batches with higher epochs (correct, no divergence)
2. Follower never updates mirrorLeaderEpoch (stuck)
3. Infinite loop of fetch -> skip -> retry (no progress)

This the exact problem I was stuck with after my initial solution.

Reply:
They should, but practice doesn't agree. Le me debug my code.

Reply:
Yes, we should figure it out, instead of workaround it without knowing the reason. Otherwise, the whole solution in the PR might base on a wrong understanding about how follower works.

Reply:
Thanks for help. Last commit should work fine.